File: backend/pkg/models/swapRequest.go
package models

import "time"

type RequestStatus string

const (
	STATUS_PENDING   RequestStatus = "pending"
	STATUS_CANCELLED RequestStatus = "cancelled"
	STATUS_APPROVED  RequestStatus = "approved"
	STATUS_DENIED    RequestStatus = "denied"
	STATUS_ARCHIVED  RequestStatus = "archived"
)

type SwapRequest struct {
	ID           string        `firestore:"id,omitempty"`
	StudentID    string        `firestore:"studentID"`
	OldSectionID string        `firestore:"oldSectionID"`
	NewSectionID string        `firestore:"newSectionID"`
	IsTemporary  bool          `firestore:"isTemporary"`
	RequestTime  time.Time     `firestore:"requestTime"`
	Reason       string        `firestore:"reason"`
	Status       RequestStatus `firestore:"status"`
	HandledBy    string        `firestore:"handledBy"`
}

File: frontend/src/model/swapRequest.ts
export type SwapRequestStatus =
  | "pending"
  | "cancelled"
  | "approved"
  | "denied"
  | "archived";

export interface SwapRequest {
  ID: string;
  studentID: string;
  oldSectionID: string;
  newSectionID: string;
  requestTime: string;
  reason: string;
  status: SwapRequestStatus;
  handledBy: string;
  assignmentID?: string;
}
